src/cpp/session/modules/SessionConnections.R

```diff
@@ -517,4 +517,36 @@ options(connectionObserver = list(
    .rs.success()
 })
 
+.rs.addJsonRpcHandler("connection_test", function(code) {
+   error <- ""
 
+   oldConnectionObserver <- getOption("connectionObserver")
+   on.exit(options(connectionObserver = oldConnectionObserver))
+
+   disconnectCalls <- list()
+
+   options(connectionObserver = list(
+      connectionOpened = function(type, host, displayName, icon = NULL, 
+                                  connectCode, disconnect, listObjectTypes,
+                                  listObjects, listColumns, previewObject, 
+                                  connectionObject, actions = NULL) {
+         disconnectCalls <<- c(disconnectCalls, disconnect)
+      },
+      connectionClosed = function(type, host, ...) {
+
+      },
+      connectionUpdated = function(type, host, hint, ...) {
+      }
+   ))
+
+   .envir <- parent.frame(2)
```

Comment:
@kevinushey based this code of https://github.com/rstudio/rstudio/blob/8b1778b0a62e989c461ca5e79eaf585f5982744d/src/cpp/session/modules/SessionCodeTools.R#L611-L630 but it looks like getActiveFrame would handle cases where one is debugging and tries to test a connection. Changed.